apps/src/templates/SignInOrAgeDialog.js

```diff
@@ -91,10 +93,16 @@ class SignInOrAgeDialog extends Component {
 
     sessionStorage.setItem(sessionStorageKey, true);
 
-    // Over 13, let them do the tutorial
-    this.setState({
-      open: false
-    });
+    // When opening a new tab, we'll have a new session (and thus show this dialog),
+    // but may still be using a storage_id for a previous user. Clear that cookie
+    // and reload
+    const cookieName = environmentSpecificCookieName('storage_id');
+    if (cookies.get(cookieName)) {
+      cookies.remove(cookieName, {path: '/', domain: 'code.org'});
```

Comment:
Are you sure the `domain` value passed here is correct?  My Chrome devtools report `.code.org` with a leading dot as the domain for the `storage_id` cookie (and for most of our cookies):

![image](https://user-images.githubusercontent.com/1615761/32455986-dfd6ab8c-c2d8-11e7-86e9-98e69f9237c0.png)

On one hand, [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) leading dots don't matter:

> Contrary to earlier specifications, leading dots in domain names are ignored. If a domain is specified, subdomains are always included.

On the other hand, the [`js-cookie` docs say](https://github.com/js-cookie/js-cookie/tree/latest#basic-usage) the following:

> _IMPORTANT! when deleting a cookie, you must pass the exact same path and domain attributes that was used to set the cookie, unless you're relying on the default attributes._

The unit test also wouldn't cover this since it stubs cookie interactions.

Reply:
I'm fairly certain I tested this locally as well, but will do so again. It sounds like this might also be something worth checking out on a couple other browsers.